api/src/main/java/edu/cornell/mannlib/vitro/webapp/search/controller/SearchFilterGroup.java

```diff
@@ -0,0 +1,65 @@
+package edu.cornell.mannlib.vitro.webapp.search.controller;
+
+import java.util.LinkedHashSet;
+
+public class SearchFilterGroup {
+
+    private String id;
+    private String label;
+    private boolean isPublic = false;
+    private boolean hidden = true;
+
+    private LinkedHashSet<String> filters = new LinkedHashSet<>();
+
+    public LinkedHashSet<String> getFilters() {
+        return filters;
+    }
+
+    public void setFilters(LinkedHashSet<String> filters) {
+        this.filters = filters;
+    }
+
+    public SearchFilterGroup(String groupId, String groupLabel) {
+        this.setId(groupId);
+        this.setLabel(groupLabel);
+    }
+
+    public void addFilterId(String filterId) {
+        if (!filters.contains(filterId)) {
+            filters.add(filterId);
+        }
```

Comment:
I think that you could just use `filters.add(filterId)` because HashSet will check whether it contains the element or not under the hood.

Reply:
Fixed